docs/backend.md

```diff
@@ -2,40 +2,271 @@
+### Registration
+Only administrators can register new users. To register a new user, the [register route](api-v1.md) has to be called.
+
+## Validation
+Every user input has to be validated before it is processed.
+
+All validations are stored in [validation.php](../api/src/validation.php) and use the [RespectValidation](https://github.com/Respect/Validation) validation engine. The stored validations have to be submitted to the [validation middleware](https://github.com/DavidePastore/Slim-Validation) of the corresponding route in [routes.php](../api/src/routes.php).
+
+**Example:**
+```php
+->add(new Validation($container['validation']['getUser']))
+```
+See more at [Routes](#route).
+
+### Translations
+In case the automatic generated error messages doesn't fit your requirements or hold sensible data like a password, you can add translations as described in [Translate errors](https://github.com/DavidePastore/Slim-Validation#translate-errors).
+
+Add the translations to [validation.php](../api/src/validation.php) and to the middleware as second parameter.
+**Example:**
+```php
+->add(new Validation(
+    $this->getContainer()['validation']['login'],
+    $this->getContainer()['validation']['loginTranslation']
+))
+```
+# Code Style
+
+It's important, that everyone complies with the following rules.
+
+**Improvements:**
+If you recognize code, which doesn't comply with these rules, just correct them. 
+
+## Optional
+In this application we use the [`Optional`](../api/src/JMP/Utils/Optional.php) very often.
+
+In php it's possible to let a method return `mixed` types (e.g. `User|bool`) or null able objects, but we decided to not use these possibilities.
+
+When a method doesn't get the expected value, in other words, the execution fails, then we return an `Optional` instead of a `null` or `false` value.
+
+### Examples
+**On success:**
+```php
+return Optional::success('the data to return, could be of any type');
+```
+**On failure:**
+```php
+return Optional::failure();
+```
+
+**Handle the returned Optional:**
+```php
+$optional = methodWhichMayFail();
+if ($optional->isSuccess()) {
+    // success
+    $data = $optional->getData();
+} else {
+//failure
+}
+```
+**Vic versa:**
+```php
+$optional = methodWhichMayFail();
+if ($optional->isFailure()) {
+    // failure
+} else {
+    // success
+    $data = $optional->getData();
+}
+```
+
+## Array Conversion
+The response object of the slim framework offers a method called `withJson`. This method converts an assoc array to JSON.
+Because the php cast functionality doesn't comply our requirements to cast model objects to associative arrays, we use the following util and interface:
```

Comment:
```suggestion
Because the php cast functionality doesn't comply with our requirements to cast model objects to associative arrays, we use the following util and interface:
```